HotelReservation/Data/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace HotelReservation.Data.Interfaces
+{
+   public interface IRepository<T> where T : class
```

Comment:
Do u really need so fat interface?
sol **I**d

Reply:
divided it into two interfaces